packages/scripts/src/ws-environment.ts

```diff
@@ -5,20 +5,36 @@ import { WsServerHost } from '@wixc3/engine-core-node';
 import type { StartEnvironmentOptions } from './types';
 import { runNodeEnvironment } from './node-environment';
 
-export async function runWSEnvironment(socketServer: Server, startEnvironmentOptions: StartEnvironmentOptions) {
-    const disposeHandlers = new Set<() => unknown>();
-    const { name } = startEnvironmentOptions;
+export function runWSEnvironment(socketServer: Server, startEnvironmentOptions: StartEnvironmentOptions) {
+    const { name, host } = startEnvironmentOptions;
     const socketServerNamespace = socketServer.of(name);
-    const host = new WsServerHost(socketServerNamespace);
-    disposeHandlers.add(() => host.dispose());
 
-    const { dispose } = await runNodeEnvironment({ ...startEnvironmentOptions, host });
-    disposeHandlers.add(() => dispose());
     return {
-        close: async () => {
-            for (const disposeHandler of [...disposeHandlers].reverse()) {
-                await disposeHandler();
-            }
+        start: async () => {
+            const wsHost = new WsServerHost(socketServerNamespace);
+
+            const runtimeEngine = await runNodeEnvironment({ ...startEnvironmentOptions, host });
+            const {
+                api: { communication },
+            } = runtimeEngine.engine.getCOM();
+            communication.registerMessageHandler(wsHost);
+            wsHost.addEventListener('message', ({ data: { from, origin } }) => {
```

Comment:
I feel like I've commented regarding this code already. is it copy/pasted? is there room for refactoring?